src/NServiceBus.Persistence.AzureStorage/SagaPersisters/MigrationSettings.cs

```diff
@@ -0,0 +1,46 @@
+namespace NServiceBus
+{
+    using Settings;
+    using Configuration.AdvancedExtensibility;
+    using Persistence.AzureStorage;
+
+    /// <summary>
+    /// Custom settings related to the outbox feature.
+    /// </summary>
+    public class MigrationSettings : ExposeSettings
+    {
+        internal MigrationSettings(SettingsHolder settings) : base(settings)
+        {
+        }
+
+        /// <summary>
+        /// By default the persistence operates in a backward compatible mode that tries to find sagas by using the secondary index property.
+        /// Once all sagas have been migrated the lookup can be disabled.
```

Comment:
Migrated from what version to what? There were already migrations that involved the secondary index. Perhaps would be worth stating migrating from X to later verions.